koku/api/report/serializers.py

```diff
@@ -324,11 +324,13 @@ def validate(self, data):
         """
         super().validate(data)
 
-        if data.get("category") and ("key_only" not in data.keys()):
-            # allow category to be passed in without group by for the tags endpoint
-            if data.get("category") and (not data.get("group_by") or not data.get("group_by").get("project")):
-                error = {"error": ("Category may not be used without a group_by project parameter")}
-                raise serializers.ValidationError(error)
+        if data.get("category") and ("key_only" in data.keys()):
```

Comment:
The `dict_keys` view isn't needed here.
```suggestion
        if data.get("category") and ("key_only" in data):
```

Reply:
Makes me wonder where else we might be doing this. 🤨 I think I may push up a separate pr to fix a few other places where we do `x in y.keys()`. Thanks for pointing this out Sam, it was some new learning for me. 🙇

Reply:
This could be fixed ad-hoc. It's probably not worth making an explicit PR just to change this.

Calling `dict.keys()` returns a `dict_keys` [view object](https://docs.python.org/3.11/library/stdtypes.html#dict-views), and then we do the test on that object. It's just an extra step that's not needed here.